### DISCUSSION TITLE
Bug? useQuery data not updated despite good onSuccess and queryFN!?

### DISCUSSION THREAD

**Discussion by tamis-laan** [TanStack/query#3294] [General]
``` javascript
// Check if token is in storage and valid
async function validateToken() {
	// Get token from storage
	var token  = await AsyncStorage.getItem('token')
	var expire = await AsyncStorage.getItem('expire')
	// Return if the token is valid
	const valid = !!token && !!expire && Date.now() < Date.parse(expire)
	console.log('validateToken', valid)
	return valid
}

// Check if loggedin
export function useLoggedin() {
	// Create query
	return useQuery('loggedin', validateToken, {
		onSuccess: () => {
			console.log('loggedin query success')
		}
	})
}
```
When I use `const {data:loggedin} = useLoggedin()` loggedin is `false`. After I login a new token is created and `loggedin` query key is invalidated. Thus `validateToken` is run and returns `true`. Then the onSuccess is triggered. However useLoggedin remains set to `false`!!
Is this a bug!?

**Comment by TkDodo** [comment]
I would need to see this in a codesandbox reproduction please.

**Comment by tamis-laan** [comment]
It happens using expo's dynamic refresh on save. If the app is reloaded the problem goes away. Because this is not obvious one can spend allot of time chasing problems that are not there.

**Reply by TkDodo** [reply]
can you reproduce it on https://snack.expo.dev/ ?

**Reply by tamis-laan** [reply]
Not really, the problem sits with the expo-cli that refreshes the app. I'm not sure how the `snack.expo.dev` website works.
